src/modules/api/user/link/link.controller.ts

```diff
@@ -1,7 +1,22 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
 import { LinkService } from './link.service';
+import { IResult } from 'src/common/interfaces/response';
 
 @Controller('users/:userId/links')
 export class LinkController {
-  constructor(private linkService: LinkService) {}
+  constructor(private readonly linkService: LinkService) {}
+  @Get()
+  async getAllLinksForUserId(@Param('userId') userId: string) {
+    const links = await this.linkService.getAllLinksOfUser(userId);
+    const response = {} as IResult;
+    if (links) {
+      response.status = 'OK';
+      response.result = links;
+    } else {
+      response.status = 'NOT OK';
+      response.error = 'Unable to get links';
+      response.result = null;
```

Comment:
Latest master's IResult implementation marked result as optional. so you can remove the 
`response.result = null;` here.